Content.Server/Whistle/Systems/WhistleSystem.cs

```diff
@@ -0,0 +1,104 @@
+using Content.Server.Whistle.Components;
+using Content.Shared.Interaction;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Movement.Components;
+using Content.Shared.Timing;
+using Robust.Server.GameObjects;
+
+namespace Content.Server.Whistle;
+
+public class WhistleSystem : EntitySystem
+{
+    [Dependency] private readonly TransformSystem _transform = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly UseDelaySystem _delaySystem = default!;
+    [Dependency] private readonly EntityLookupSystem _entityLookup = default!;
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<WhistleComponent, BeforeRangedInteractEvent>(OnBeforeInteract);
+        SubscribeLocalEvent<WhistleComponent, UseInHandEvent>(OnUseInHand);
+    }
```

Comment:
This should just use useinhand.

Reply:
wdym by just use useinhand?

Maybe you mean that i should remove first event and left second?